src/containers/Tenant/Diagnostics/TopQueries/TopQueriesData.tsx

```diff
@@ -0,0 +1,71 @@
+import {ResponseError} from '../../../../components/Errors/ResponseError';
+import {ResizeableDataTable} from '../../../../components/ResizeableDataTable/ResizeableDataTable';
+import {TableWithControlsLayout} from '../../../../components/TableWithControlsLayout/TableWithControlsLayout';
+import {topQueriesApi} from '../../../../store/reducers/executeTopQueries/executeTopQueries';
+import type {KeyValueRow} from '../../../../types/api/query';
+import type {EPathType} from '../../../../types/api/schema';
+import {cn} from '../../../../utils/cn';
+import {isSortableTopQueriesProperty} from '../../../../utils/diagnostics';
+import {useAutoRefreshInterval, useTypedSelector} from '../../../../utils/hooks';
+import {QUERY_TABLE_SETTINGS} from '../../utils/constants';
+import {isColumnEntityType} from '../../utils/schema';
+
+import {TOP_QUERIES_COLUMNS, TOP_QUERIES_COLUMNS_WIDTH_LS_KEY} from './getTopQueriesColumns';
+import i18n from './i18n';
+
+const b = cn('kv-top-queries');
+
+interface Props {
+    database: string;
+    onRowClick: (row: any) => void;
+    type?: EPathType;
+}
+
+export const TopQueriesData = ({database, onRowClick, type}: Props) => {
+    const [autoRefreshInterval] = useAutoRefreshInterval();
+    const filters = useTypedSelector((state) => state.executeTopQueries);
+    const {currentData, isFetching, error} = topQueriesApi.useGetTopQueriesQuery(
+        {
+            database,
+            filters,
+        },
+        {pollingInterval: autoRefreshInterval},
+    );
+    const loading = isFetching && currentData === undefined;
+    const {result: data} = currentData || {};
+
+    const rawColumns = TOP_QUERIES_COLUMNS;
+    const columns = rawColumns.map((column) => ({
+        ...column,
+        sortable: isSortableTopQueriesProperty(column.name),
+    }));
+
+    const handleRowClick = (row: KeyValueRow) => {
+        return onRowClick(row.QueryText as string);
+    };
+
+    if (error && !data) {
+        return (
+            <TableWithControlsLayout.Table>
+                <ResponseError error={error} />
+            </TableWithControlsLayout.Table>
+        );
+    }
+
+    if (!data || isColumnEntityType(type)) {
+        return <TableWithControlsLayout.Table>{i18n('no-data')}</TableWithControlsLayout.Table>;
+    }
+
+    return (
+        <TableWithControlsLayout.Table loading={loading}>
+            <ResizeableDataTable
+                columnsWidthLSKey={TOP_QUERIES_COLUMNS_WIDTH_LS_KEY}
+                columns={columns}
+                data={data}
+                settings={QUERY_TABLE_SETTINGS}
+                onRowClick={handleRowClick}
+                rowClassName={() => b('row')}
+            />
+        </TableWithControlsLayout.Table>
```

Comment:
Loading state will never be shown, you will show either cached error or no-data. Also now it's possible to forget `TableWithControlsLayout.Table` wrapper in some of conditions.

Could you please implement the same approach like in `RunningQueriesData`?